models/allArticles.model.js

```diff
@@ -0,0 +1,62 @@
+const db = require("../db/connection");
+
+function selectAllArticles() {
+  return db
+    .query(
+      `
+      SELECT 
+  articles.author,
+  articles.title,
+  articles.article_id,
+  articles.topic,
+  articles.created_at,
+  articles.votes,
+  articles.article_img_url,
+  COUNT(comments.article_id) AS comment_count
+FROM 
+  articles
+  LEFT JOIN
+  comments ON articles.article_id = comments.article_id
+GROUP BY
+  articles.author,
+  articles.title,
+  articles.article_id,
+  articles.topic,
+  articles.created_at,
+  articles.votes,
+  articles.article_img_url
+ORDER BY
+  articles.created_at DESC;
+
+    `
+    )
+    .then((result) => {
+      const articles = result.rows;
+      if (articles.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: "No articles found.",
+        });
+      }
+      return articles.map((article) => {
+        return {
+          author: article.author,
+          title: article.title,
+          article_id: article.article_id,
+          topic: article.topic,
+          date: article.created_at,
+          votes: article.votes,
+          image: article.article_img_url,
+          comment_count: parseInt(article.comment_count),
```

Comment:
It would be better to not have this logic here for the sake of efficiency, if you want comment_count to be a number we can do that in our SQL query using CAST() 
More info here: https://www.w3schools.com/sql/func_sqlserver_cast.asp